internal/controller/apiextensions/usage/reconciler.go

```diff
@@ -320,17 +320,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		if u.Spec.ReplayDeletion != nil && *u.Spec.ReplayDeletion && used.GetAnnotations() != nil {
 			if policy, ok := used.GetAnnotations()[usage.AnnotationKeyDeletionAttempt]; ok {
 				// We have already recorded a deletion attempt and want to replay deletion, let's delete the used resource.
-				go func() {
+				go func(ctx context.Context) {
 					// We do the deletion async and after some delay to make sure the usage is deleted before the
 					// deletion attempt. We remove the finalizer on this Usage right below, so, we know it will disappear
 					// very soon.
-					time.Sleep(2 * time.Second)
-					log.Info("Replaying deletion of the used resource", "apiVersion", used.GetAPIVersion(), "kind", used.GetKind(), "name", used.GetName(), "policy", policy)
-					// We cannot use the context from the reconcile function since it will be cancelled after the reconciliation.
-					if err = r.client.Delete(context.Background(), used, client.PropagationPolicy(policy)); err != nil {
-						log.Info("Error when replaying deletion of the used resource", "apiVersion", used.GetAPIVersion(), "kind", used.GetKind(), "name", used.GetName(), "err", err)
+					select {
+					case <-ctx.Done():
+						log.Info("Context cancelled, not replaying deletion of the used resource", "apiVersion", used.GetAPIVersion(), "kind", used.GetKind(), "name", used.GetName(), "policy", policy)
+						return
+					case <-time.After(2 * time.Second):
+						log.Info("Replaying deletion of the used resource", "apiVersion", used.GetAPIVersion(), "kind", used.GetKind(), "name", used.GetName(), "policy", policy)
+						// We cannot use the context from the reconcile function since it will be cancelled after the reconciliation.
+						if err = r.client.Delete(ctx, used, client.PropagationPolicy(policy)); err != nil {
+							log.Info("Error when replaying deletion of the used resource", "apiVersion", used.GetAPIVersion(), "kind", used.GetKind(), "name", used.GetName(), "err", err)
+						}
 					}
-				}()
+				}(ctx)
```

Comment:
Per the note above:

> // We cannot use the context from the reconcile function since it will be cancelled after the reconciliation.

Not relying on context here was intentional indeed.

Reply:
😅 oops, noted: more reading and less committing 😅